### PR TITLE
Add support for language server in PHAR

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: scoper.inc.php [hunk 1]
```diff
@@ -102,18 +102,9 @@ function ($filePath, $prefix, $contents) {
...
-                $prefix . '\Psalm\Plugin\\',
-                'Psalm\Plugin\\',
-                $contents
-            );
-            return $ret;
-        },
     ],
     'whitelist' => [
         \Composer\Autoload\ClassLoader::class,
-        'Psalm\Plugin\*',
```
Comment (weirdan) [inline scoper.inc.php]: Wouldn't these changes cause problems running plugins with phar-packaged Psalm?

Comment (muglug) [reply scoper.inc.php]: Probably - I was just seeing errors when running on Psalm’s own codebase, and assumed the errors were universal. I’ll remove tomorrow

**Comment by muglug** [comment]
I added in c1434ee2120c16090d6aaef42f244e6aaa90ca9b, but thanks for your help!

File: scoper.inc.php (hunk 1 above)
Comment (muglug) [reply scoper.inc.php]: Do plugins work with the Phar at all?

**Comment by greg0ire** [comment]
No, thanks for _your_ help!

File: scoper.inc.php (hunk 1 above)
Comment (weirdan) [reply scoper.inc.php]: Partially. For example, PHPUnit plugin loads, but will probably fail on some versions where it accesses internal classes. Plugins can work with phar, given enough care both from the plugin and Psalm's sides. As long as phar build does not mangle any class names plugin may use it should be working just fine.

Comment (muglug) [reply scoper.inc.php]: Yeah, but the plugin hooks demand non-namespaced things like `Psalm\StatementsSource`

Comment (weirdan) [reply scoper.inc.php]: - Not every plugin may have hooks (doctrine plugin is still stubs-only, but that's going to change)
- My original idea was to provide interfaces in `Plugin` namepsace for everything plugins may need to depend on

**Pull request closed without merging**